Problem building on macos-15 (arm64)

Seems to be related to Highway.
```
[46/46] Linking target _apytypes.cpython-313-darwin.so
  FAILED: _apytypes.cpython-313-darwin.so
  c++  -o _apytypes.cpython-313-darwin.so _apytypes.cpython-313-darwin.so.p/src_apycfloatarray.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfloatarray_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfixed.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfixed_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfixedarray.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfixedarray_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfloat.cc.o _apytypes.cpython-313-darwin.so.p/src_apycfloat_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfixed.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfixed_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfixedarray.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfixedarray_iterator.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfixedarray_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfloat.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfloat_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfloatarray.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfloatarray_iterator.cc.o _apytypes.cpython-313-darwin.so.p/src_apyfloatarray_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apytypes_common.cc.o _apytypes.cpython-313-darwin.so.p/src_apytypes_context_wrapper.cc.o _apytypes.cpython-313-darwin.so.p/src_apytypes_mp.cc.o _apytypes.cpython-313-darwin.so.p/src_apytypes_simd.cc.o _apytypes.cpython-313-darwin.so.p/src_apytypes_wrapper.cc.o -flto -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,dynamic_lookup -bundle -Wl,-undefined,dynamic_lookup subprojects/nanobind-2.4.0/libnanobind.a subprojects/highway/libhwy.a
  ld: warning: linking module flags 'SDK Version': IDs have conflicting values ('[2 x i32] [i32 15, i32 0]' from subprojects/highway/libhwy.a[5](hwy_per_target.cc.o) with '[2 x i32] [i32 15, i32 2]' from ld-temp.o)
  ld: warning: linking module flags 'SDK Version': IDs have conflicting values ('[2 x i32] [i32 15, i32 0]' from subprojects/highway/libhwy.a[7](hwy_targets.cc.o) with '[2 x i32] [i32 15, i32 2]' from ld-temp.o)
  LLVM ERROR: Cannot select: 0x10732be50: nxv4i32 = AArch64ISD::SUNPKLO 0x107321550
    0x107321550: v2i64 = llvm.aarch64.neon.ushl Constant:i32<658>, 0x10733ed70, 0x10734be90
      0x107365b10: i32 = Constant<658>
      0x10733ed70: v2i64,ch = load<(load (s128) from %ir.44, align 8, !tbaa !489)> 0x11cfc9d10, 0x107316fa0, undef:i64
        0x107316fa0: i64 = add 0x10e85d240, Constant:i64<-32>
          0x10e85d240: i64,ch = CopyFromReg 0x11cfc9d10, Register:i64 %19
            0x107352750: i64 = Register %19
          0x1073200f0: i64 = Constant<-32>
        0x107342600: i64 = undef
      0x10734be90: v2i64 = AssertZext 0x10736dd40, ValueType:ch:i32
        0x10736dd40: v2i64,ch = CopyFromReg 0x11cfc9d10, Register:v2i64 %13
          0x10731f6e0: v2i64 = Register %13
  In function: _ZN4simd5N_SVE28_hwy_vector_shift_div_signedEPxPKxS3_jm
  clang++: error: unable to execute command: Abort trap: 6
  clang++: error: linker command failed due to signal (use -v to see invocation)
  ninja: build stopped: subcommand failed.
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  Preparing metadata (pyproject.toml): finished with status 'error'
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /Library/Frameworks/Python.framework/Versions/3.13/bin/python /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /var/folders/r4/ph658m2d3vs0vj00g75q0zyr0000gn/T/tmp3wr1zts3
  cwd: /Users/runner/work/apytypes/apytypes
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -26,7 +26,7 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.10", "3.11", "3.12", "3.13", "3.13t", "3.14-dev", "3.14t-dev"]
-        os: [ubuntu-latest, ubuntu-24.04-arm, windows-latest, macos-13, macos-14] # macos-15, windows-11-arm,
+        os: [ubuntu-latest, ubuntu-24.04-arm, windows-latest, macos-13, macos-14, macos-15] # windows-11-arm,
         numpy-version: [">=1.26.4,<2", ">=2.1"]
         exclude:
           # Only Python 3.13 and later for windows-11-arm
@@ -78,7 +78,7 @@ jobs:
     - name: Install NumPy
       run: |
         python -m pip install 'numpy${{ matrix.numpy-version }}'
-    - name: Install NumPy
+    - name: Install development NumPy
       if: contains(matrix.python-version, 'dev')
       run: |
         python -m pip install --pre --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --upgrade --only-binary=:all: 'numpy${{ matrix.numpy-version }}'
@@ -86,12 +86,11 @@ jobs:
       run: |
         git fetch --tags
         python -m pip install .[test] -v
-    - name: Test with pytest
-      if: matrix.python-version == '3.13t'
+    - name: Test with pytest (free-threaded)
+      if: contains(matrix.python-version, 't')
       run: |
         export PYTHON_GIL=0
         pytest --color=yes lib/test
     - name: Test with pytest
-      if: matrix.python-version != '3.13t'
       run: |
         pytest --color=yes lib/test

